### PR TITLE
Fixes unique constraint error of batchId when scheduling emails

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

This PR creates a unique batchId for each message that is scheduled so the database can be successfully updated. It also adds a missing response for `api/cron/workflows/scheduleEmailReminders` and `api/cron/workflows/scheduleSMSReminders`.

Fixes the following issue:
If we scheduled several messages, the following update failed as the batchId has a unique constraint (but it wasn't unique): 

Screenshot 2022-07-26 at 23 01 24

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Looks good apart from the concerns I raised. Blocking because one of them can be a potential bug

File: apps/web/ee/pages/api/cron/workflows/scheduleEmailReminders.ts [hunk 1]
```diff
@@ -107,30 +105,39 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
...
             from: senderEmail,
             subject: emailContent.emailSubject,
             text: emailContent.emailBody.text,
             html: emailContent.emailBody.html,
-            batchId: batchIdResponse[1].batch_id,
+            batchId: batchId,
             sendAt: dayjs(reminder.scheduledDate).unix(),
           });
+
+          await prisma.workflowReminder.updateMany({
```
[the remark this answers is not in the document]
Comment (zomars) [reply apps/web/ee/pages/api/cron/workflows/scheduleEmailReminders.ts:125]: Yup. This won't work as expected

**Review by zomars — COMMENTED** [review]
I'll delegate my approval to @hariombalhara Overall nice work @CarinaWolli

File: apps/web/ee/pages/api/cron/workflows/scheduleEmailReminders.ts [hunk 2]
```diff
@@ -107,30 +105,39 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
...
+            where: {
+              id: reminder.id,
```
[the remark this answers is not in the document]
Comment (CarinaWolli) [reply apps/web/ee/pages/api/cron/workflows/scheduleEmailReminders.ts:127]: The ID is unique, I changed updateMany to update. Made the changes also in scheduleSMSReminders.ts

**Comment by CarinaWolli** [comment]
Implemented the feedback and resolved merge conflicts, it's ready for another round of review 🙏 @zomars @hariombalhara

File: packages/features/ee/workflows/api/scheduleSMSReminders.ts [hunk 3]
```diff
@@ -48,7 +48,7 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
 
   const dateInSevenDays = dayjs().add(7, "day");
 
-  unscheduledReminders.forEach(async (reminder) => {
+  for (const reminder of unscheduledReminders) {
```
Comment (zomars) [inline packages/features/ee/workflows/api/scheduleSMSReminders.ts:51]: I'm assuming we don't care about these updates being run out of order right? If that's so then LGTM. We can either:
- [Run them in serial](https://decembersoft.com/posts/promises-in-serial-with-array-reduce/) or...
- Run them in parallel with Promise.all

**Review by zomars — APPROVED** [review]

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**